File: carddemo-java/carddemo-core/src/main/java/com/carddemo/core/domain/Account.java
package com.carddemo.core.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;

/**
 * Account entity mapped from COBOL copybook CVACT01Y.
 * Original VSAM file: AWS.M2.CARDDEMO.ACCTDATA.PS (KSDS, 300-byte records)
 * Primary key: ACCT-ID PIC 9(11)
 */
@Entity
@Table(name = "account", indexes = {
        @Index(name = "idx_account_group_id", columnList = "group_id"),
        @Index(name = "idx_account_active_status", columnList = "active_status")
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Account {

    @Id
    @Column(name = "acct_id")
    private Long acctId;

    @NotNull
    @Column(name = "active_status", length = 1)
    private String activeStatus;

    @Column(name = "current_balance", precision = 12, scale = 2)
    private BigDecimal currentBalance;

    @Column(name = "credit_limit", precision = 12, scale = 2)
    private BigDecimal creditLimit;

    @Column(name = "cash_credit_limit", precision = 12, scale = 2)
    private BigDecimal cashCreditLimit;

    @Column(name = "open_date")
    private LocalDate openDate;

    @Column(name = "expiration_date")
    private LocalDate expirationDate;

    @Column(name = "reissue_date")
    private LocalDate reissueDate;

    @Column(name = "current_cycle_credit", precision = 12, scale = 2)
    private BigDecimal currentCycleCredit;

    @Column(name = "current_cycle_debit", precision = 12, scale = 2)
    private BigDecimal currentCycleDebit;

    @Column(name = "address_zip", length = 10)
    private String addressZip;

    @Column(name = "group_id", length = 10)
    private String groupId;
}

File: backend/CardDemo.Domain/Accounts/Account.cs
namespace CardDemo.Domain.Accounts;

/// <summary>
/// Account master record, ported from ACCOUNT-RECORD (app/cpy/CVACT01Y.cpy, RECLN 300).
/// VSAM ACCTDAT KSDS KEYS(11 0) = ACCT-ID. Amounts are PIC S9(10)V99; dates are X(10) yyyy-MM-dd.
/// </summary>
public class Account
{
    public required string AccountId { get; set; }
    public required string ActiveStatus { get; set; }
    public decimal CurrentBalance { get; set; }
    public decimal CreditLimit { get; set; }
    public decimal CashCreditLimit { get; set; }
    public DateOnly? OpenDate { get; set; }
    public DateOnly? ExpirationDate { get; set; }
    public DateOnly? ReissueDate { get; set; }
    public decimal CurrentCycleCredit { get; set; }
    public decimal CurrentCycleDebit { get; set; }
    public required string AddressZip { get; set; }
    public required string GroupId { get; set; }
}
